systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaNodePoolST.java

```diff
@@ -308,13 +317,164 @@ void testKafkaNodePoolBrokerIdsManagementUsingAnnotations(ExtensionContext exten
+        //waitForPodsReady(testStorage.getNamespaceName(), testStorage.getKafkaSelector(), scaledUpBrokerReplicaCount, false);
+
+        // assert that new changed configuration is present in pods under Kafka Node Pool A
+        verifyResourcesConfigurationPropagatedToKnpPods(testStorage, nodePoolNameA, kafkaJvmXmxConfig, kafkaLimitMemoryQuantityConfigModified);
+
+        LOGGER.info("Modifying configuration relevant or irrelevant based on role of given KafkaNodePools");
+
+        final Pod podFromKnpC = PodUtils.getPodsByPrefixInNameWithDynamicWait(testStorage.getNamespaceName(), testStorage.getClusterName() + "-" + nodePoolNameC).get(0);
+        final String podCName = podFromKnpC.getMetadata().getName();
+        final String knpCConfigHashAnnotationValue = podFromKnpC.getMetadata().getAnnotations().get(Constants.NODE_BROKER_CONFIG_HASH_ANNOTATION);
+        final String knpCStrimziPodSetName = KafkaNodePoolUtils.getStrimziPodSetName(testStorage.getClusterName(), nodePoolNameC);
+
+        // replacing config which is not relevant to control (also it is part of dynamic config, therefore should not trigger Rolling Update)
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(), k -> {
+            k.getSpec().getKafka().getConfig().put("compression.type", "gzip");
+        }, testStorage.getNamespaceName());
+
+        StrimziPodSetUtils.waitForPrevailedPodAnnotationKeyValuePairs(testStorage.getNamespaceName(), Constants.NODE_BROKER_CONFIG_HASH_ANNOTATION, knpCConfigHashAnnotationValue, knpCStrimziPodSetName, podCName);
+
+        // replacing config relevant only to controllers
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(), k -> {
+            k.getSpec().getKafka().getConfig().put("max.connections", "456");
+        }, testStorage.getNamespaceName());
+
+        StrimziPodSetUtils.waitUntilPodAnnotationChange(testStorage.getNamespaceName(), podCName, knpCStrimziPodSetName, Constants.NODE_BROKER_CONFIG_HASH_ANNOTATION, knpCConfigHashAnnotationValue);
+    }
+
+    /**
+     * Verifies that the specified configurations are correctly propagated to the Kafka Node Pool (KNP) pods.
+     *
+     * <p>This method checks the memory limit and JVM heap options of the first pod belonging to the given node pool.
+     *
+     * @param testStorage        the storage object containing test-related information.
+     * @param nodePoolName       the name of the KNP to verify.
+     * @param expectedJvmXmxLimit the expected JVM '-Xmx' heap size limit.
+     * @param expectedMemoryLimit the expected memory limit ('resource.limits.memory') for the pod.
+     *
+     * @throws AssertionError if configurations obtained from Pods do not match the expected values.
+     */
+    public static void verifyResourcesConfigurationPropagatedToKnpPods(TestStorage testStorage, String nodePoolName, String expectedJvmXmxLimit, String expectedMemoryLimit) {
```

Comment:
There is no need to have this method static and also public...
```suggestion
    private void verifyResourcesConfigurationPropagatedToKnpPods(TestStorage testStorage, String nodePoolName, String expectedJvmXmxLimit, String expectedMemoryLimit) {
```